pytest.ini

```diff
@@ -9,5 +9,9 @@
 # deprecation warnings and gives time for plugins that don't use the deprecated
 # API to propagate, especially for plugins packaged as an external snap, before
 # we release breaking changes.
+#
+# The current warnings being ignored are:
+# 1) Our own warning about the deprecated acme.crypto_util.make_self_signed_cert function.
 filterwarnings =
     error
+    ignore:make_self_signed_cert is deprecated:DeprecationWarning
```

Comment:
a little tempted to suggest just removing `GenMakeSelfSignedCertTest` now and not needing this warning since we're not using it anymore, but I'm fine with leaving it now and removing it along with the code later, as well.

Reply:
how about my most recent commit?

it fixes up tests to actually test for the warning and allows us to not ignore the warning globally so if we ever try to write a new call to make_self_signed_cert, we'll see the warning. it also lets us keep tests around until we actually remove the function. i personally think an approach like this is ideal whenever we deprecate functionality

if you don't like this for some reason tho, just lmk whatever approach you prefer and i'll implement that